skfda/tests/test_covariances.py

```diff
@@ -1,106 +1,265 @@
+    """Fixture for getting a covariance kernel function."""
+    return request.param
+
+
+@pytest.fixture(
+    params=[
+        cov.Brownian(),
+        cov.WhiteNoise(),
+    ],
+)
+def covariances_raise_fixture(request: Any) -> Any:
+    """Fixture for getting a covariance kernel that raises a ValueError."""
+    return request.param
+
+
+@pytest.fixture
+def fdatabasis_data() -> Tuple[FDataBasis, FDataBasis]:
+    """Fixture for getting fdatabasis objects."""
+    basis = MonomialBasis(
+        n_basis=2,
+        domain_range=(-2, 2),
+    )
+
+    fd1 = FDataBasis(
+        basis=basis,
+        coefficients=[
+            [1, 0],
+            [1, 2],
+        ],
+    )
+
+    fd2 = FDataBasis(
+        basis=basis,
+        coefficients=[
+            [0, 1],
+        ],
+    )
+
+    return fd1, fd2
+
+
+@pytest.fixture
+def multivariate_data() -> np.array:
+    """Fixture for getting multivariate data."""
+    return np.linspace(-1, 1, 1000)[:, np.newaxis]
+
+
+@pytest.fixture(
+    params=[
+        (cov.Linear,
+         {
+             "variance": [1, 2],
+             "intercept": [3, 4],
+         },
+         ),
+        (cov.Polynomial,
+         {
+             "variance": [2],
+             "intercept": [0, 2],
+             "slope": [1, 2],
+             "degree": [1, 2, 3],
+         },
+         ),
+        (cov.Exponential,
+         {
+             "variance": [1, 2],
+             "length_scale": [0.5, 1, 2],
+         },
+         ),
+        (cov.Gaussian,
+         {
+             "variance": [1, 2],
+             "length_scale": [0.5, 1, 2],
+         },
+         ),
+        (cov.Matern,
+         {
+             "variance": [2],
+             "length_scale": [0.5],
+             "nu": [0.5, 1, 1.5, 2.5, 3.5, np.inf],
+         },
+         ),
+    ],
+)
+def covariance_and_params(request: Any) -> Any:
+    """Fixture to load the covariance functions."""
+    return request.param
+
+
+##############################################################################
+# Tests
+##############################################################################
+
+
+def test_covariances(
+    fetch_functional_data: FDataGrid,
+    covariances_fixture: cov.Covariance,
+) -> None:
+    """Check that invalid parameters in fit raise exception."""
+    fd = fetch_functional_data
+    cov_kernel = covariances_fixture
+
+    # Also test that it does not fail
+    res1 = cov_kernel(fd, fd)
+    res2 = cov_kernel(fd)
+
+    np.testing.assert_allclose(
+        res1,
+        res2,
+        atol=1e-7,
+    )
+
+
+def test_raises(
+    fetch_functional_data: FDataGrid,
+    covariances_raise_fixture: Any,
+) -> None:
+    """Check that it raises a ValueError exception."""
+    fd = fetch_functional_data
+    cov_kernel = covariances_raise_fixture
+
+    np.testing.assert_raises(
+        ValueError,
+        cov_kernel,
+        fd,
+    )
+
+
+def test_fdatabasis_example_linear(
+    fdatabasis_data: Tuple[FDataBasis, FDataBasis],
```

Comment:
The name of the fixture, `fdatabasis_data`, seems to indicate that this is **arbitrary** data in `FDataBasis` representation. However, the results are **not** arbitrary, but done for particular numbers. I suggest to create a different fixture with a more explicit name (and a docstring indicating its purpose), or to generate the data inside the test, if it is test-specific. It would be good if you could seek more information about best practices with fixtures in this regard.

Reply:
I included a more explanatory docstring for the fixture.

Reply:
The problem is that there is a strong dependency between the expected result and the value returned by the fixture. I do not think that is a good idea. Maybe `fdatabasis_data` should not be a fixture?